internal/storage/sql/common/util.go

```diff
@@ -0,0 +1,30 @@
+package common
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+
+	ferrors "go.flipt.io/flipt/errors"
+)
+
+// decodePageToken is a utility function which determines the `PageToken` based on
+// input.
+func decodePageToken(pageToken string) (PageToken, error) {
+	var token PageToken
+
+	tok, err := base64.StdEncoding.DecodeString(pageToken)
+	if err != nil {
+		return token, ferrors.ErrInvalidf("pageToken is not valid: %q", pageToken)
+	}
+
+	if err := json.Unmarshal(tok, &token); err != nil {
+		if _, ok := ferrors.As[*json.SyntaxError](err); ok {
+			return token, ferrors.ErrInvalidf("pageToken is not valid: %q", pageToken)
+		}
+
+		return token, fmt.Errorf("decoding page token: %w", err)
```

Comment:
Was just thinking about this for a second. I wonder if we should just always return invalid token here? Otherwise, there will be some invalid shape tokens which cause a 500 and some that cause a 4XX. However, seems that any kind of invalid unmarshalling here should be treated equally.
We could pass down a logger though and warn out what the detail is for the server operator to see in the logs.
```suggestion
		logger.Warn("invalid page token provided", zap.Error(err))
		
		return token, ferrors.ErrInvalidf("pageToken is not valid: %q", pageToken)
```

Reply:
@GeorgeMac Yeah that sounds good. It makes page token being invalid be inconsistent in every case. The only thing I can think of is an unexpected error when marshalling that should never occur which would warrant the 500. But Its a pretty flat function anyway.

Thanks!